deps/build.jl

```diff
@@ -33,21 +32,34 @@ end
 libfftw = library_dependency(libfftw_name, aliases=makealiases(libfftw_name))
 libfftwf = library_dependency(libfftwf_name, aliases=makealiases(libfftwf_name))
 
-provides(AptGet, "libfftw3-double3", [libfftw], os=:Linux)
-provides(AptGet, "libfftw3-single3", [libfftwf], os=:Linux)
-provides(Pacman, "fftw", [libfftw, libfftwf], os=:Linux)
-provides(Zypper, "libfftw3_threads3", [libfftw, libfftwf], os=:Linux)
-provides(Yum, "fftw-libs", [libfftw, libfftwf], os=:Linux)
-provides(BSDPkg, "fftw3", [libfftw, libfftwf], os=:FreeBSD)
+url = "https://github.com/ararslan/fftw-builder/releases/download/v\$FFTW_VER/libfftw-\$FFTW_VER"
+
+isdir(joinpath(@__DIR__, "usr", "lib")) || mkpath(joinpath(@__DIR__, "usr", "lib"))
 
-if iswindows()
-    using WinRPM
-    provides(WinRPM.RPM, "libfftw3-3", [libfftw, libfftwf], os=:Windows)
-elseif isapple()
-    using Homebrew
-    provides(Homebrew.HB, "fftw", [libfftw, libfftwf], os=:Darwin)
+if Sys.isapple()
+    provides(Binaries, URI("\$url-osx-x86_64.tar.gz"), [libfftw, libfftwf],
+             SHA="e405d0006ddd5f229dc9fd4eee5bd3ee74ca3302173eb088306ae559ae9de01e",
+             unpacked_dir="usr/lib", os=:Darwin)
+elseif Sys.islinux() && Sys.ARCH in [:x86_64, :i686]
+    sha = if Sys.ARCH === :x86_64
+        "72b15f0dcc20d346d3e2fe73a9068a3cc1198705083b92e19d76aa5140968f0b"
+    elseif Sys.ARCH === :i686
+        "fcb36713fcddc2aa8c16c8df021482f502c185bcabe447614868e70a9933f019"
+    end
+    provides(Binaries, URI("\$url-linux-\$(Sys.ARCH).tar.gz"), [libfftw, libfftwf],
+             unpacked_dir="usr/lib", SHA=sha, os=:Linux)
+elseif Sys.iswindows() && Sys.ARCH in [:x86_64, :i686]
+    sha = if Sys.ARCH === :x86_64
+        "e0376b43bcd161762726f683b4404844388fc2b8997ba86405555f844d390f37"
+    elseif Sys.ARCH === :i686
+        "67b8084cfed5f240aff70890369111f38fed624e8fbae58719c3d1ef58ca02ff"
+    end
+    provides(Binaries, URI("\$url-win-\$(Sys.ARCH).zip"), [libfftw, libfftwf],
+             unpacked_dir="usr/lib", SHA=sha, os=:Windows)
```

Comment:
I would suggest something a little more ordered, like what [Tom did for `Sundials.jl`](https://github.com/JuliaDiffEq/Sundials.jl/blob/7b679d931198cc5c646ca29390ff3ac9879b67cb/deps/build.jl).  Maybe with a check to see if there just aren't any platforms available at all.